contrib/generate-ds20.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+# pylint: disable=invalid-name,missing-docstring
+#
+# Copyright (C) 2022 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+# pylint: disable=consider-using-f-string
+
+import sys
+import argparse
+import configparser
+import base64
+from typing import List
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-b",
+        "--bufsz",
+        type=int,
+        help="Buffer size in bytes",
+    )
+    parser.add_argument("-i", "--instance-id", type=str, help="Device instance ID")
+    parser.add_argument("filename", action="store", type=str, help="Quirk filename")
+
+    args = parser.parse_args()
+    if not args.bufsz:
+        parser.print_help()
+        sys.exit(1)
+
+    config = configparser.ConfigParser()
+    config.optionxform = str
+    try:
+        config.read(args.filename)
+    except configparser.MissingSectionHeaderError:
+        print("not a quirk file")
+        sys.exit(1)
+
+    # fall back to the default if there is only one device in the quirk file
+    if not args.instance_id:
+        sections = config.sections()
+        if len(sections) != 1:
+            print("multiple devices found, use --instance-id to choose")
```

Comment:
What do you think about printing them in the case of  this error message?

Reply:
Something like:

    multiple devices found, use --instance-id to choose between USB\VID_273F&PID_1000, USB\VID_04D8&PID_F8DA, USB\VID_273F&PID_1001, USB\VID_273F&PID_1004, USB\VID_273F&PID_1005, USB\VID_273F&PID_1007, USB\VID_273F&PID_1006

-- or a bullet list? That might be easier to copy-paste.

Reply:
Bullet list all the way. What I'm thinking here was all about the copy paste indeed!

Reply:
$ ./contrib/generate-ds20.py plugins/colorhug/colorhug.quirk  --bufsz 235
	Multiple devices found, use --instance-id to choose between:
	 • USB\VID_273F&PID_1000
	 • USB\VID_04D8&PID_F8DA
	 • USB\VID_273F&PID_1001
	 • USB\VID_273F&PID_1004
	 • USB\VID_273F&PID_1005
	 • USB\VID_273F&PID_1007
	 • USB\VID_273F&PID_1006

Reply:
That looks good.  I'll leave one more usability comment separate